### PR TITLE
Adds basic docker config

### PR DESCRIPTION
**Pull request by adaam2**

### CODE REVIEW

File: backend/config/initializers/redis.rb [hunk 1]
```diff
@@ -1 +1 @@
-REDIS = Redis.new
+REDIS = Redis.new(host: ENV['IS_DOCKER'] ? 'redis' : 'localhost')
```
Comment (adaam2) [inline backend/config/initializers/redis.rb:1]: Guess we still want this to work in a non docker env

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+version: '3'
+
+networks:
+  app-tier:
+    driver: bridge
```
Comment (adaam2) [inline docker-compose.yml:5]: Probs don't need the network anymore.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+services:
+  redis:
+    image: redis:latest
+    command: ["redis-server", "--bind", "0.0.0.0", "--port", "6379"]
```
Comment (adaam2) [inline docker-compose.yml:10]: Probs don't need this given i'm using interpolation.

**Review by alexstoick — COMMENTED** [review]
I would add volumes in the `docker-compose` otherwise you need to rebuild the image everything you change something, which is suboptimal.

**Pull request merged**